docker/Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Build the consensus-specs image using the builder pattern.
+# This image is base on the Debian slim distribution.
+FROM python:3.11.0-slim-bullseye as base
+
+
+FROM base as builder
+RUN mkdir /consensus-specs
+WORKDIR /consensus-specs
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt update && apt install -y git
+RUN python -m pip install --upgrade pip
+
+COPY . .
+
+RUN pip install -r requirements-dev.txt -r requirements_preinstallation.txt
```

Comment:
1. `requirements_preinstallation.txt` should be installed before `requirements-dev.txt`
2. Can `python -m pip install -e .[test]` replace `pip install -r requirements-dev.txt` so that we don't need to maintain `requirements-dev.txt` file?

Reply:
> 1. `requirements_preinstallation.txt` should be installed before `requirements-dev.txt`
You are right indeed.

> 2. Can `python -m pip install -e .[test]` replace `pip install -r requirements-dev.txt` so that we don't need to maintain `requirements-dev.txt` file?
Indeed good idea !